src/control/security/grpc_authorization_test.go

```diff
@@ -53,65 +57,152 @@ func TestCommonNameToComponent(t *testing.T) {
 	}
 }
 
-func TestHasAccess(t *testing.T) {
-	allComponents := [4]Component{ComponentUndefined, ComponentAdmin, ComponentAgent, ComponentServer}
-	testCases := []struct {
-		testname         string
-		correctComponent Component
-		method           string
-	}{
-		{"StoragePrepare", ComponentAdmin, "/ctl.MgmtCtl/StoragePrepare"},
-		{"StorageScan", ComponentAdmin, "/ctl.MgmtCtl/StorageScan"},
-		{"StorageFormat", ComponentAdmin, "/ctl.MgmtCtl/StorageFormat"},
-		{"SystemQuery", ComponentAdmin, "/ctl.MgmtCtl/SystemQuery"},
-		{"SystemStop", ComponentAdmin, "/ctl.MgmtCtl/SystemStop"},
-		{"SystemResetFormat", ComponentAdmin, "/ctl.MgmtCtl/SystemResetFormat"},
-		{"SystemStart", ComponentAdmin, "/ctl.MgmtCtl/SystemStart"},
-		{"NetworkScan", ComponentAdmin, "/ctl.MgmtCtl/NetworkScan"},
-		{"FirmwareQuery", ComponentAdmin, "/ctl.MgmtCtl/FirmwareQuery"},
-		{"FirmwareUpdate", ComponentAdmin, "/ctl.MgmtCtl/FirmwareUpdate"},
-		{"Join", ComponentServer, "/mgmt.MgmtSvc/Join"},
-		{"LeaderQuery", ComponentAdmin, "/mgmt.MgmtSvc/LeaderQuery"},
-		{"PoolCreate", ComponentAdmin, "/mgmt.MgmtSvc/PoolCreate"},
-		{"PoolDestroy", ComponentAdmin, "/mgmt.MgmtSvc/PoolDestroy"},
-		{"PoolQuery", ComponentAdmin, "/mgmt.MgmtSvc/PoolQuery"},
-		{"PoolSetProp", ComponentAdmin, "/mgmt.MgmtSvc/PoolSetProp"},
-		{"PoolGetACL", ComponentAdmin, "/mgmt.MgmtSvc/PoolGetACL"},
-		{"PoolOverwriteACL", ComponentAdmin, "/mgmt.MgmtSvc/PoolOverwriteACL"},
-		{"PoolUpdateACL", ComponentAdmin, "/mgmt.MgmtSvc/PoolUpdateACL"},
-		{"PoolDeleteACL", ComponentAdmin, "/mgmt.MgmtSvc/PoolDeleteACL"},
-		{"PoolExclude", ComponentAdmin, "/mgmt.MgmtSvc/PoolExclude"},
-		{"PoolDrain", ComponentAdmin, "/mgmt.MgmtSvc/PoolDrain"},
-		{"PoolReintegrate", ComponentAdmin, "/mgmt.MgmtSvc/PoolReintegrate"},
-		{"PoolEvict", ComponentAdmin, "/mgmt.MgmtSvc/PoolEvict"},
-		{"PoolExtend", ComponentAdmin, "/mgmt.MgmtSvc/PoolExtend"},
-		{"GetAttachInfo", ComponentAgent, "/mgmt.MgmtSvc/GetAttachInfo"},
-		{"BioHealthQuery", ComponentAdmin, "/mgmt.MgmtSvc/BioHealthQuery"},
-		{"SmdListDevs", ComponentAdmin, "/mgmt.MgmtSvc/SmdListDevs"},
-		{"SmdListPools", ComponentAdmin, "/mgmt.MgmtSvc/SmdListPools"},
-		{"ListPools", ComponentAdmin, "/mgmt.MgmtSvc/ListPools"},
-		{"DevStateQuery", ComponentAdmin, "/mgmt.MgmtSvc/DevStateQuery"},
-		{"StorageSetFaulty", ComponentAdmin, "/mgmt.MgmtSvc/StorageSetFaulty"},
-		{"ListContainers", ComponentAdmin, "/mgmt.MgmtSvc/ListContainers"},
-		{"PrepShutdownRanks", ComponentServer, "/mgmt.MgmtSvc/PrepShutdownRanks"},
-		{"StopRanks", ComponentServer, "/mgmt.MgmtSvc/StopRanks"},
-		{"PingRanks", ComponentServer, "/mgmt.MgmtSvc/PingRanks"},
-		{"ResetFormatRanks", ComponentServer, "/mgmt.MgmtSvc/ResetFormatRanks"},
-		{"StartRanks", ComponentServer, "/mgmt.MgmtSvc/StartRanks"},
+func TestSecurity_ComponentHasAccess(t *testing.T) {
```

Comment:
Originally these were subtests so we could print out an easier to understand test name for the user in the event of test failure. This is why we had a clear test name field in the original array. I had it the way you have now in the first version of my PR that introduced the tests but discussion lead us to have the clearer names. Do we still want to do this? You might be able to have clearer names by just taking the token after the last / in the method name.

Reply:
I've updated the test so that the displayed names don't have the prefix.